### PR TITLE
Remove V1 wrapper: SoftwareVersion.

### PR DESCRIPTION
**Pull request by denisshevchenko**
<p align="right">#119</p>

# Overview

- [x] Type `SoftwareVersion` doesn't use `V1` wrapper.
- [x] Golden test both for old and new types passed.

# Comments

### CODE REVIEW

**Comment by KtorZ** [comment]
@denisshevchenko Some styling issues unfortunately.

File: src/Cardano/Wallet/API/V1/Types.hs [hunk 1]
```diff
@@ -1982,6 +1984,41 @@ instance BuildableSafeGen WalletAndTxHistory where
...
+instance Arbitrary WalletSoftwareVersion where
+    arbitrary = WalletSoftwareVersion <$> arbitrary
+
+-- | NOTE: There is 'ToSchema' instance for 'V1 SoftwareVersion' type, but it's defined
+-- in 'cardano-sl-chain' package and we cannot touch it for now. So we have to define
+-- here some 'ToSchema'-instance for 'WalletSoftwareVersion'.
+instance ToSchema WalletSoftwareVersion where
+    declareNamedSchema _ =
+        pure $ NamedSchema (Just "WalletSoftwareVersion") $ mempty
+            & type_ .~ SwaggerObject
```
Comment (KtorZ) [inline src/Cardano/Wallet/API/V1/Types.hs:2030]: So... If we weren't defining any custom instances for `SoftwareVersion`.. What was `V1` used for ... ? Sounds like we could just remove it entirely without going for a newtype.

Comment (denisshevchenko) [reply src/Cardano/Wallet/API/V1/Types.hs:2030]: I've tried it already, but it requires `ToSchema` for `SoftwareVersion`, it doesn't compile without it. But we cannot define orphan instances for `Core`-types, right?

Comment (KtorZ) [reply src/Cardano/Wallet/API/V1/Types.hs:2030]: but where do the previous `ToSchema` instance came from... ?

Comment (denisshevchenko) [reply src/Cardano/Wallet/API/V1/Types.hs:2030]: We have `instance ToSchema (V1 SoftwareVersion)` in `cardano-sl-chain` package.

**Review by KtorZ — APPROVED** [review]

**Comment by KtorZ** [comment]
@denisshevchenko This conflicts with the PR on `Coin`.

**Comment by denisshevchenko** [comment]
@KtorZ Fixed.

**Pull request merged**